src/core/MOM_barotropic.F90

```diff
@@ -4885,9 +4926,11 @@ subroutine barotropic_init(u, v, h, eta, Time, G, GV, US, param_file, diag, CS,
   ALLOC_(CS%IdyCv(CS%isdw:CS%iedw,CS%jsdw-1:CS%jedw)) ; CS%IdyCv(:,:) = 0.0
   ALLOC_(CS%dy_Cu(CS%isdw-1:CS%iedw,CS%jsdw:CS%jedw)) ; CS%dy_Cu(:,:) = 0.0
   ALLOC_(CS%dx_Cv(CS%isdw:CS%iedw,CS%jsdw-1:CS%jedw)) ; CS%dx_Cv(:,:) = 0.0
+  allocate(CS%IareaT_OBCmask(isd:ied,jsd:jed))
   do j=G%jsd,G%jed ; do i=G%isd,G%ied
     CS%IareaT(i,j) = G%IareaT(i,j)
     CS%bathyT(i,j) = G%bathyT(i,j)
+    CS%IareaT_OBCmask(i,j) = CS%IareaT(i,j)
```

Comment:
`CS%IareaT_OBCmask(i,j)` is being initialized for the normal-sized halo regions, but it is a wide-halo array, so halo update is needed after it is set, presumably as a part of the group halo update at about line 4944.